### PR TITLE
ft(Creating-Product): Allow a seller to create/add a product

### PR DESCRIPTION
**Pull request by Aldot-02**
## ℹ️ Description

As a Seller

- I should be able to add/create items for sale
- This page should only be accessible to sellers, i.e authenticated
- Such that the stock can be seen by any requesting client
- Handle all responses and errors from the backend

## 🧪 How can this be tested?

This should be tested on the vendor dashboard, under the route of vendor/create that can be accessed by clicking the Add product button.

## 📸 Screenshots

Screenshot 2024-06-27 at 12 06 47

Screenshot 2024-07-09 at 14 47 14

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [X] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by martinemahirwe — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

File: src/components/ToastMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        </button>
+        <button
+          className="bg-black text-white p-2 rounded-md hover:bg-gray-800"
+          onClick={() => {
+            toast.dismiss();
+            if (onProceed) onProceed();
+          }}
+        >
+          Proceed to Stock
+        </button>
```
Comment (SergeInGithub) [inline src/components/ToastMessage.tsx]: We have a button component that can be used here

File: src/pages/vendorpages/addProducts.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+              className="w-full p-2 border border-gray-300 rounded-md h-24 focus:outline-none focus:shadow-outline"
+            ></textarea>
+          </div>
+          <button
+            type="submit"
+            className="w-full bg-black text-white p-2 rounded-md hover:bg-gray-800"
+            disabled={isCreating}
+          >
+            {isCreating ? 'Saving changes...' : 'Save changes'}
+          </button>
```
Comment (SergeInGithub) [inline src/pages/vendorpages/addProducts.tsx]: We have a button component that can be used or improved if it's not dynamic enough

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Nice work so far. Look into some reviews below @Aldot-02

**Review by martinemahirwe — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

**Review by martinemahirwe — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**